### DISCUSSION TITLE
Change the maximum size of request and response

### DISCUSSION THREAD

**Discussion by cyrus937** [libp2p/rust-libp2p#5710] [Q&A] [answered]
Hello !
I would like to increase the maximum size of requests and responses in the request_response protocol.
Please, does anyone know how I could do this?

I've seen that it's set to :
```rust
     /// Max request size in bytes
    const REQUEST_SIZE_MAXIMUM: u64 = 1024 * 1024;
    /// Max response size in bytes
    const RESPONSE_SIZE_MAXIMUM: u64 = 10 * 1024 * 1024;
```

**Comment by romac** [comment, marked as answer]
The only way I have found to work around that is to implement your own [`Codec`](https://docs.rs/libp2p/latest/libp2p/request_response/trait.Codec.html), using either the CBOR or JSON codecs as reference.

**Reply by cyrus937** [reply]
Thanks !

**Comment by dariusc93** [comment]
I do have a [branch](https://github.com/dariusc93/rust-libp2p/commit/91c26048e0f949dc4417e9514469fbf28aca218e) with the change that I meant to make into a PR but this is the change you could make.

**Reply by moofone** [reply]
Are you still planning the PR? I patched it locally also but would be nice to have it included

Edit: I see it - thanks :)

**Reply by vladfaust** [reply]
This is the PR: https://github.com/libp2p/rust-libp2p/pull/5792.

**Comment by uniartdes** [comment]
thank you
